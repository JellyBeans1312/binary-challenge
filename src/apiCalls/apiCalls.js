export const getEvents = async () => {
  try {
    const options = { 
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }

    const response = await fetch('https://api.predicthq.com/v1/events/', options)
    const result = await response.json()
    return result.results
  } catch(error) {
    console.log(error);
  }
}

export const searchEvent = async (searchParam) => {
  try {
    const options = {
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }
    
    const response = await fetch(`https://api.predicthq.com/v1/events/?q=${searchParam}/`, options)
    const result = await response.json()
    return result.results

  } catch(error) {
    throw new Error(error)
  }
}

export const getAddress = async (lat, long) => {
  try {
    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${long}&key=AIzaSyDPPlNyjATG8sQnfsdg45ln7kUoyFH24m0`)
    const result = await response.json();
    return result.results.map(result => {
      return result.formatted_address
    });
  } catch(error) {
    console.log(error)
  }
}