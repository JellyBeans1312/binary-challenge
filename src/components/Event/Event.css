.event {
  background-image: url('../../images/event-background.jpg');
  background-size: cover;
  height: 200px;
  width: 400px;
  font-size: 1.2rem;
}