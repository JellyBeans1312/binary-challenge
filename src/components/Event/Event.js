import React, { Component } from 'react';
import { getAddress } from '../../apiCalls/apiCalls'

export class Event extends Component {
  constructor(props) {
    super(props);
    this.state = {
      formattedAddress: ''
    }
  }
  componentDidMount = async () => {
    const { location } = this.props
    console.log(location)
    console.log(location[1].toString())
    const formattedAddress = await getAddress(location[1].toString(), location[0].toString())
    console.log(formattedAddress)
    this.setState({ formattedAddress })
  }
  
  trimDescription = () => {
    const { description } = this.props
  }

  render() {
    const { title, description, address } = this.props 
    const { formattedAddress } = this.state
    return (
      <section>
        <h4>{!title && 'No available title'}</h4>
        <h4>{title}</h4>
        <p>{!description && 'No available Description'}</p>
        <p>{description}</p>
        <p>{!formattedAddress && 'No available Location'}</p>
        <p>{formattedAddress}</p>
        <p>{!address && 'No available Address'}</p>
        <p>{address}</p>

      </section>
    )
  }
}