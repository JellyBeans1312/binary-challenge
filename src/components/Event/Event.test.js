import React from 'react';
import { shallow } from 'enzyme';
import { Event } from './Event';
import { getAddress } from '../../apiCalls/apiCalls';


jest.mock('../../apiCalls/apiCalls')

describe('Event', () => {
  let wrapper;

  const mockLocation = [
    432123, -12345
  ]
  beforeEach(() => {
    wrapper = shallow(
      <Event
      title='Event'
      description='Its an event'
      category='concert'
      location={mockLocation}
      />
    )
    getAddress.mockImplementation(() => {
      return Promise.resolve([{address: 'something 123 something else drive'}])
    });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  }); 

  it('should setState to trimDescription', () => {
    wrapper.instance().trimDescription()
    expect(wrapper.state('trimDesc')).toEqual('Its an event...')
  });

  it('should call getAddress on mounting', () => {
    expect(getAddress).toHaveBeenCalled()
  });
})