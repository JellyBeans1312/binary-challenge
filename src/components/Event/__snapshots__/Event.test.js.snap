// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should match snapshot 1`] = `
<article
  className="event"
>
  <h4 />
  <h4>
    Event
  </h4>
  <p />
  <p>
    No available Address
  </p>
  <p />
</article>
`;
